Change to use single test results file for jsc tests

facetothefate opened the pull request:
#### bdd01b8335f05aab55cafe0677c2f6df565d2129
<pre>
Change to use single test results file for jsc tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=264908">https://bugs.webkit.org/show_bug.cgi?id=264908</a>
<a href="https://rdar.apple.com/118482056">rdar://118482056</a>

Reviewed by Ryan Haddad.

Currently we generate a test status file for each jsc test we run, and then
we will use a find command for remote devices to get the content of the each
file.

This find command is time consuming, it may take forever for low end embedded
device. This patch will make the test result ouput to a single file, then we copy
the file back the host, open it locally, and process the results line by line.
It will save a lot of time for the script to report the final status.

* Tools/Scripts/run-jsc-stress-tests:
* Tools/Scripts/webkitruby/jsc-stress-test-writer-default.rb:
* Tools/Scripts/webkitruby/jsc-stress-test/executor.rb:

Canonical link: <a href="https://commits.webkit.org/272663@main">https://commits.webkit.org/272663@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a506c6fb0fb379566319213145c4998140a70fb4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32568 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11316 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34416 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35134 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29440 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33434 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13665 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8510 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/28964 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33002 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9537 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29100 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8313 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8447 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29041 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36470 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/27937 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29600 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29464 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34570 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/32632 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8577 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6530 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32429 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10252 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/39070 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7581 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9189 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8249 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9170 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aoikonomopoulos commented on Tools/Scripts/run-jsc-stress-tests:
```diff
@@ -2813,6 +2813,16 @@ def copyBundleToRemote(remoteHost)
+    hostResultFolder = getLocalRemoteResultPath(remoteHost)
+    Dir.mkdir hostResultFolder unless hostResultFolder.directory?
+    mysys(["scp"] + SSH_OPTIONS_DEFAULT + (remoteHost.identity_file_path ? ["-i", remoteHost.identity_file_path] : []) + ["-P", remoteHost.port.to_s, "#{remoteHost.user}@#{remoteHost.host}:#{remoteHost.remoteDirectory}/#{$outputDir.basename}/.runner/#{STATUS_FILE}", "#{getLocalRemoteResultPath(remoteHost).to_s}/#{STATUS_FILE}" ])
+end
```
There's nothing wrong with this per-se, but I wonder if it would be more straightforward to use `sshRead` along with `cat` in `getStatusMap` instead of invoking `scp` directly here and using a temporary file.

facetothefate replied:
Using driver host to read will be more fast, since the remote device is often an embedded device with limited CPU and IO speed.

aoikonomopoulos replied:
Just to be clear, what I was suggesting was using the equivalent of `ssh host cat file` instead of `scp host:file .`, simply because we already have a mechanism for the former. So I don't think the embedded device would need to use more CPU time. Not that important though.

aoikonomopoulos commented on Tools/Scripts/webkitruby/jsc-stress-test-writer-default.rb:
```diff
@@ -252,7 +252,7 @@ def reproScriptCommand
 
     def statusCommand(status)
-        "echo #{$runUniqueId} $exitCode #{status} > #{statusFile}"
+        "echo #{@index} #{$runUniqueId} $exitCode #{status} >> #{statusFile}"
```
Hmm, I wonder if this is going to result in an ever-growing `statusFile` for CI systems. One way to work around this would be to include the `$runUniqueId` in the filename and remove it when done (may leave a stale file behind though). Alternatively, remove the `statusFile` before starting?

facetothefate replied:
Now it will clear the temp folder every time when it finished processing the results.  See [Tools/Scripts/run-jsc-stress-tests:2911]

ryanhaddad reviewed (approved):
r=me with the comments from Angelos addressed

facetothefate committed:
Change to use single test results file for jsc tests
https://bugs.webkit.org/show_bug.cgi?id=264908
rdar://118482056

Reviewed by Ryan Haddad.

Currently we generate a test status file for each jsc test we run, and then
we will use a find command for remote devices to get the content of the each
file.

This find command is time consuming, it may take forever for low end embedded
device. This patch will make the test result ouput to a single file, then we copy
the file back the host, open it locally, and process the results line by line.
It will save a lot of time for the script to report the final status.

* Tools/Scripts/run-jsc-stress-tests:
* Tools/Scripts/webkitruby/jsc-stress-test-writer-default.rb:
* Tools/Scripts/webkitruby/jsc-stress-test/executor.rb:

Canonical link: https://commits.webkit.org/272663@main
